File: frontend/src/types/Review.ts
import { Product } from "./Product";
import { User } from "./User";

export interface Review {
  _id: string;
  product: Product;
  user: User;
  rating: number;
  comment: string;
  createdAt: string;
  updatedAt: string;
}

export interface ReviewResponse {
  success: boolean;
  count: number;
  data: Review[];
  message?: string;
  error?: string;
}

File: Mobileptit/app/src/main/java/com/uteshop/app/data/model/Review.java
package com.uteshop.app.data.model;

import com.google.gson.JsonElement;
import com.google.gson.annotations.SerializedName;

public class Review {
    @SerializedName("_id")public String id;
    public JsonElement product;
    public JsonElement user;
    public Integer rating;
    public String comment;
    public String createdAt;
    public String updatedAt;
}
